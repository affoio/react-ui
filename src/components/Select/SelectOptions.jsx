import React from 'react';
import styled from 'styled-components';
import { getMultipliedIndent } from '../../theme/selectors';
import colors from '../../colors';

const SelectOptionsStyled = styled.div`
  max-height: ${({ theme }) => getMultipliedIndent(theme, 61)};
  padding: ${({ theme }) => getMultipliedIndent(theme, 2)} 0;
  margin-top: ${({ theme, inline }) => {
    const margin = getMultipliedIndent(theme, 2);
    if (!inline) {
      return `-${margin}`;
    }
    return margin;
  }};
  overflow: auto;
  box-shadow: 0 0 15px 0 rgba(0, 0, 0, 0.2);
  background-color: ${colors.white};
`;

const SelectOptions = React.forwardRef((props, ref) => <SelectOptionsStyled {...props} ref={ref} />);

export default SelectOptions;
