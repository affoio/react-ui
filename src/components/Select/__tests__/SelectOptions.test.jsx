import React from 'react';
import renderer from 'react-test-renderer';
import SelectOptions from '../SelectOptions';

describe('<SelectOptions', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SelectOptions />);

    expect(tree).toMatchSnapshot();
  });
});
