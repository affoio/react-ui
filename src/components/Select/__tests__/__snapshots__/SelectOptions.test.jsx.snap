// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectOptions renders correctly 1`] = `
.c0 {
  max-height: 244px;
  padding: 8px 0;
  margin-top: -8px;
  overflow: auto;
  box-shadow: 0 0 15px 0 rgba(0,0,0,0.2);
  background-color: #ffffff;
}

<div
  className="c0"
/>
`;
